### PR TITLE
added service principal authentication version to nodejs

### PR DESCRIPTION
**Pull request by wruwami**
i want to use service principal authentication in API Samples - NodeJS

### CODE REVIEW

File: API Samples - NodeJS/utils.js [hunk 1]
```diff
@@ -30,12 +38,29 @@ function validateConfig(){
         return "AuthorityUrl is empty. Please fill valid AuthorityUrl under config.json";
     }
 
-    if (!config.username || !config.username.trim()){
-        return "Username is empty. Please fill Power BI username in config.json";
-    }
+    if(config.authenticationType === 'MasterUser')
+    {
```
Comment (anant-k-singh) [inline API Samples - NodeJS/utils.js]: Open brace in the same line of _if condition_ for consistent coding style

File: API Samples - NodeJS/utils.js [hunk 2]
```diff
@@ -10,6 +10,14 @@ function validateConfig(){
     var guid = require('guid');
     var config = require(__dirname + '/config.json');
 
+    if(!config.authenticationType){
+        return "authenticationType is empty. please choose PowerUser or ServicePrincipal in config.json";
```
Comment (anant-k-singh) [inline API Samples - NodeJS/utils.js]: **P**lease choose **MasterUser** or ServicePrincipal in config.json

File: API Samples - NodeJS/authentication.js [hunk 3]
```diff
@@ -29,17 +29,33 @@ async function getAuthenticationToken() {
...
     // use user credentials and appId to get an aad token
-    let promise = () => { return new Promise(
-        (resolve, reject) => {
-            context.acquireTokenWithUsernamePassword(config.resourceUrl, config.username, config.password, config.appId , function(err, tokenResponse) {
-                if (err) reject(err);
-                resolve(tokenResponse);
-            })
-        });
-    };
+    if (config.authenticationType == "MasterUser")
```
Comment (anant-k-singh) [inline API Samples - NodeJS/authentication.js]: Using string equality is recommended

File: API Samples - NodeJS/authentication.js [hunk 4]
```diff
@@ -29,17 +29,33 @@ async function getAuthenticationToken() {
...
+    {
```
Comment (anant-k-singh) [inline API Samples - NodeJS/authentication.js]: Open brace in the same line of _if condition_ for consistent coding style

File: API Samples - NodeJS/utils.js (hunk 1 above)
Comment (lipsotiko) [reply API Samples - NodeJS/utils.js]: https://standardjs.com can help with consistency

File: API Samples - NodeJS/authentication.js (hunk 3 above)
Comment (lipsotiko) [reply API Samples - NodeJS/authentication.js]: https://standardjs.com can help with syntax, === vs ==

File: API Samples - NodeJS/authentication.js [hunk 5]
```diff
@@ -29,17 +29,33 @@ async function getAuthenticationToken() {
     var cas = JSON.parse(casjson);
     https.globalAgent.options.ca = cas;
 
-    var context = new AuthenticationContext(authorityUrl);
-
+    var promise = undefined;
```
Comment (parth-007) [inline API Samples - NodeJS/authentication.js:32]: this is promiseFactory, not promise.

File: API Samples - NodeJS/authentication.js (hunk 3 above)
Comment (parth-007) [reply API Samples - NodeJS/authentication.js:34]: You should check the case too.

File: API Samples - NodeJS/authentication.js [hunk 6]
```diff
@@ -29,17 +29,33 @@ async function getAuthenticationToken() {
...
+        var context = new AuthenticationContext(authorityUrl);
+        promise = () => { return new Promise(
+            (resolve, reject) => {
+                context.acquireTokenWithUsernamePassword(config.resourceUrl, config.username, config.password, config.appId , function(err, tokenResponse) {
+                    if (err) reject(err);
+                    resolve(tokenResponse);
+                })
+            });
+        };
+    }
```
Comment (parth-007) [inline API Samples - NodeJS/authentication.js:45]: else if should be here for consistent coding style.

File: API Samples - NodeJS/utils.js (hunk 1 above)
Comment (parth-007) [reply API Samples - NodeJS/utils.js:42]: Check for the case

File: API Samples - NodeJS/utils.js [hunk 7]
```diff
@@ -30,12 +38,29 @@ function validateConfig(){
...
+            return "Username is empty. Please fill Power BI username in config.json";
+        }
 
-    if (!config.password || !config.password.trim()){
-        return "Password is empty. Please fill password of Power BI username in config.json";
+        if(!config.password || !config.password.trim()){
+            return "Password is empty. Please fill password of Power BI username in config.json";
+        }
+    }
+    else if(config.authenticationType === 'ServicePrincipal')
```
Comment (parth-007) [inline API Samples - NodeJS/utils.js:51]: Check for the case

**Comment by ali-hamud** [comment]
We are publishing new samples. For nodeJS we are publishing a sample which works with service principal as well.
Closing this PR. Thank you.

**Pull request closed without merging**